### PR TITLE
Configuration system implementation

### PR DESCRIPTION
**Pull request by adrienbanse**
When initializing `ecologits` with 
```python
Ecologits.init(
    providers = provider_list, 
    electricity_mix_zone = "BEL"
)
```
the static object `Config` (in `ecologits.config`) takes the corresponding attributes, and is module-wide accessible. 

Remains to 
- [x] ~~Implement a verbose argument~~ Implement proper logging
- [x] Write tests
- [x] Write corresponding doc

### CODE REVIEW

**Comment by adrienbanse** [comment]
@samuelrince I didn't implement a verbose argument but a proper logging
Not finding a model or a mix are at the `WARNING` level
The user can then decide the level himself or herself.

File: ecologits/_ecologits.py
Comment (samuelrince) [inline ecologits/_ecologits.py:1]: Instead of having a config file that could be altered externally, I'd rather have it inside the EcoLogits class. The instance and definition of configs can be changed from the EcoLogits object only. 

Something like this:

```python
from dataclasses import dataclass, field

class EcoLogits:
	
	@dataclass
	class _Config:
		electricity_mix_zone: str = field(default="WOR")
		provider: list[str] = field(default_factory=list)
	
	config = _Config()	# Init with defaults
	
	@staticmethod
	def init(providers = ..., electricity_mix_zone = ...):
		...
		
		EcoLogits.config.electricity_mix_zone = electricity_mix_zone
		EcoLogits.config.providers += providers
		...
```

Then whenever we want to access the config internally we can do the following:

```python
from ecologits._ecologits import EcoLogits

def func(...):
	electricity_mix = EcoLogits.config.electricity_mix_zone	#Access config value
```

What do you think of these changes?

File: ecologits/tracers/utils.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
+import logging
 from typing import Optional
 
 from ecologits.electricity_mix_repository import electricity_mixes
 from ecologits.impacts.llm import compute_llm_impacts
 from ecologits.impacts.modeling import Impacts, Range
 from ecologits.model_repository import models
 
+logger = logging.getLogger(__name__)
```
Comment (samuelrince) [inline ecologits/tracers/utils.py:9]: I would define a logger in a log.py and import it wherever we need logging.

```python
import logging

logger = logging.getLogger("ecologits")
```

File: ecologits/tracers/utils.py [hunk 2]
```diff
@@ -33,8 +36,7 @@ def llm_impacts(
 
     model = models.find_model(provider=provider, model_name=model_name)
     if model is None:
-        # TODO: Replace with proper logging
-        print(f"Could not find model `{model_name}` for {provider} provider.")
+        logger.warning(f"Could not find model `{model_name}` for {provider} provider.")
```
Comment (samuelrince) [inline ecologits/tracers/utils.py:39]: I think the "warning" level is excessive and will still be spammy as exposed in #65. We should probably have an "errors" attribute in the "Impacts" object in the response instead of a logging message.

For now, I would just switch to `logger.debug`.

File: ecologits/tracers/utils.py [hunk 3]
```diff
@@ -43,8 +45,7 @@ def llm_impacts(
 
     electricity_mix = electricity_mixes.find_electricity_mix(zone=electricity_mix_zone)
     if electricity_mix is None:
-        # TODO: Replace with proper logging
-        print(f"Could not find electricity mix `{electricity_mix_zone}` in the ADEME database")
+        logger(f"Could not find electricity mix `{electricity_mix_zone}` in the ADEME database")
```
Comment (samuelrince) [inline ecologits/tracers/utils.py:48]: Same logger.debug

File: pyproject.toml [hunk 4]
```diff
@@ -49,6 +49,7 @@ aiohttp = { version = "^3.9.3", optional = true }
 minijinja = { version = "^1.0.16", optional = true }
 google-generativeai = { version = "^0.7.0", optional = true }
 litellm = {version="^1.41.23", optional = true}
+logging = "^0.4.9.6"
```
Comment (samuelrince) [inline pyproject.toml:52]: Logging is part of the standard python library, so we do not need to install it or have it as a dependency,

File: tests/test_litellm.py
Comment (samuelrince) [inline tests/test_litellm.py:1]: Mistral models didn't work?

File: tests/test_tracer.py [hunk 5]
```diff
@@ -32,3 +35,51 @@ def test_init_openai_only():
...
+            {
+                "role": "user",
+                "content": "Hello, Claude",
+            }
+        ],
+        model="claude-3-opus-20240229",
+    )
+
+    assert message_under_belgian_mix.impacts.energy.value.min != message_under_french_mix.impacts.energy.value.min
+
```
Comment (samuelrince) [inline tests/test_tracer.py:85]: Some remarks on this test:

1. I would use OpenAI with [deterministic output](https://cookbook.openai.com/examples/reproducible_outputs_with_the_seed_parameter) (i.e. the seed set to `0`). This will make sure we have the same output for the same input parameters, and thus make the two request impacts comparable.
2. As it is written, the test should be failing. Changing the electricity mix does not impact the energy consumption. We should test for GWP impacts instead. The test is currently not failing because we compare equality of floats that are not exactly the same 0.008138263895 ≈ 0.0081090964725
3. I would also change the comparison to FRA < WOR impacts instead of just checking inequality.

**Review by samuelrince — CHANGES_REQUESTED** [review]

File: ecologits/_ecologits.py
Comment (adrienbanse) [reply ecologits/_ecologits.py:1]: Sure, to me it's also good! I took inspiration from `openllmetry`, I thought having a `config.py` file was customary

File: ecologits/tracers/utils.py (hunk 1 above)
Comment (adrienbanse) [reply ecologits/tracers/utils.py:9]: Isn't it exactly the same? I thought the logger was determined by the name only

File: tests/test_tracer.py (hunk 5 above)
Comment (adrienbanse) [reply tests/test_tracer.py:85]: Completely agree! Went too fast on this one

File: ecologits/tracers/utils.py (hunk 1 above)
Comment (samuelrince) [reply ecologits/tracers/utils.py:9]: Yes, indeed from a practical standpoint it does not change anything, but I see 3 advantages :

1. Do not display library submodule in logs

If we define the logger with `logging.getLogger(__name__)` in the root of the package or in a submodule, we will get difference logging sources.

```
[DATE] [LEVEL] ecologits: message
[DATE] [LEVEL] ecologits.submodule: message
```

I think it is preferred to only log with "ecologits" unless if it makes sense to have submodules. In our case, the end user only sees the EcoLogits object and should not know about tracers, for instance. Plus, it makes filtering easier for the end user, just target "ecologits" and that's it.

2. It removes the ambiguity or possibility of errors

Removes the possibility of having a typo in the name of the logger, for instance. (though it shouldn't happen with good reviewing, we can eliminate the risk by configuring it once).

3. If we define some customizations on the logger we only do it once.

For instance, if we want to automatically deduplicate messages, we could add a specific function for that. Once it is implemented in the logger, we no longer need to care whether it is available or not in other parts of the library.

**Review by samuelrince — APPROVED** [review]

**Comment by samuelrince** [comment]
I let you @adrienbanse double-check the changes in the documentation, if you are ok with it, you can rebase and merge! :)
